crates/hir-ty/src/infer/closure.rs

```diff
@@ -232,7 +232,8 @@ pub enum CaptureKind {
 pub struct CapturedItem {
     pub(crate) place: HirPlace,
     pub(crate) kind: CaptureKind,
-    pub(crate) span: MirSpan,
+    /// The inner vec is the stacks; the outer vec is for each capture reference.
+    span_stacks: SmallVec<[SmallVec<[MirSpan; 3]>; 3]>,
```

Comment:
Might be good to change the outer len to 1 to keep the size down of this struct (though I am not sure if this is used enough to matter).

Reply:
Also what are the "stacks", what do the stacks here represent?